src/js/components/ResultItem.jsx

```diff
@@ -51,9 +51,6 @@ export class ResultItem extends Component {
             const statusGlyph = getGlyphFor(resultClean);
             this.setState({resultClean, statusGlyph});
         }
-        if (props.expanded!=this.props.expanded) {
```

Comment:
On props change yes, that is the intend of the change. 

The component use the state to keep track on that events see toggleExpanded

```
const expanded = !this.state.expanded;
```

The problem is that above code is, it triggers  this.toggleExpanded(); and not check the new value of the expanded from nextProps but simply toggles the current expanded in the state. Now in our use case we have expanded via onExpand and clicking on a line in the log will add the hash to the current location. That will change the prop of the step in which the log resist and will collapse it which is not what it should do.

...but writing this explanation I think I have an alternative approach. Cheers @kzantow  for being 
![rubber-duck_0](https://cloud.githubusercontent.com/assets/596701/16803388/5e58ef9a-4906-11e6-8485-d82386c66d1f.jpg)